Disable claim victory on direct challenge or rated classical

ThomasKluiters opened the pull request:
This will make it so that it is no longer possible to force resignation of your opponent if either:
- The game was originally from a challenge
- The game is rated classical

Users will no longer see the 'claim victory' pop up and the backend won't allow any force resignations if either of these two are true.

@ornicar I made the assumption that you meant "friend game" OR "rated classical", if this should have been an AND, I'll change it.

Fixes #4866

ornicar commented:
I meant it as "AND". We certainly need to handle the classical lobby players who rage quit.

Apart from that the PR looks good and ready to land.

ThomasKluiters commented:
> I meant it as "AND". We certainly need to handle the classical lobby players who rage quit.
> 
> Apart from that the PR looks good and ready to land.

Fixed!

Also modified some function names for some consistency.

ThomasKluiters commented:
Also: note to self: never use GitHub directly for changing files.. Messed my commits up big time

ornicar commented:
uh, yeah, also github doesn't compile or run your changes.

greg-finley-ck commented on modules/round/src/main/RoundDuct.scala:
```diff
@@ -74,10 +74,12 @@ private[round] final class Round(
-          case true if !pov.game.variant.insufficientWinningMaterial(pov.game.board, pov.color) => finisher.rageQuit(pov.game, Some(pov.color))
-          case true => finisher.rageQuit(pov.game, None)
-          case _ => fuccess(List(Event.Reload))
+        pov.forceResignable ?? {
```
Should `forceResignable` be brought up to the prior line?

```
case ResignForce(playerId) => handle(playerId) { pov =>
      (pov.game.resignable && !pov.game.hasAi && pov.game.hasClock && pov.forceResignable)** ?? {
```

ThomasKluiters replied:
Semantically there is no difference, personally I prefer the extra line to make it a bit more clear "can we resign this game at all?" => "can we FORCE resign this game?" => ...
